src/Dropdown.tsx

```diff
@@ -265,11 +281,12 @@ export function Dropdown ({
   onSelect,
   onRemove
 }: DropdownProps) {
-  const [cta, ctaSet] = React.useState(
-    get(items.filter(i => i.value === value)[0], 'label') ||
-      placeholder ||
-      'Please select'
-  )
+  const [cta, ctaSet] = React.useState(getCTA({ items, value, placeholder }))
+
+  // will only call ctaSet if computedCTA changes
+  React.useEffect(() => {
+    ctaSet(getCTA({ items, value, placeholder }))
+  }, [items, value, placeholder])
```

Comment:
@angelac17 broke this down further: 
- synchronously sets initial state
- updates state if pertinent values change

Reply:
This look great! Thanks for putting out an update so quickly🥇